src/jit/simdcodegenxarch.cpp

```diff
@@ -619,8 +619,8 @@ CodeGen::genSIMDIntrinsicInit(GenTreeSIMD* simdNode)
             // data section using movss/sd.  Similarly if op1 is a memory op we
             // would have loaded it using movss/sd.  Movss/sd when loading a xmm reg
             // from memory would zero-out upper bits. In these cases we can
-            // avoid explicitly zero'ing out targetReg.
-            bool zeroInitRequired = !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
+            // avoid explicitly zero'ing out targetReg if targetReg and op1Reg are the same.
+            bool zeroInitRequired = (targetReg != op1Reg) || !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
```

Comment:
I see one issue in genSIMDScalarMove() that is using ins_Store(type) instead of using ins_Copy(type) for reg-to-reg move when zeroInit is false.  But not sure whether that will explain this bug.  Using ins_Copy(type) will result in using movaps for reg-to-reg moves.